src/snowflake/snowpark/_internal/compiler/query_generator.py

```diff
@@ -218,11 +218,11 @@ def do_resolve_with_resolved_children(
 
         elif isinstance(logical_plan, WithQueryBlock):
             resolved_child = resolved_children[logical_plan.children[0]]
-            # record the CTE definition of the current block
-            if logical_plan.name not in self.resolved_with_query_block:
-                self.resolved_with_query_block[
-                    logical_plan.name
-                ] = resolved_child.queries[-1]
+            # record the CTE definition of the current block and update the query when
```

Comment:
For clarity, I think it should it be `or` instead of `and`, right?
```suggestion
            # record the CTE definition of the current block or update the query when
```